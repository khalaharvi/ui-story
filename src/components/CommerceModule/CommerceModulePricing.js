import React from 'react';
import PropTypes from 'prop-types';

import { cx } from './CommerceModule';

const CommerceModulePricing = ({
  currency,
  price,
  priceRaw
}) => (
  <dl className={cx('pricing')}>
    { priceRaw && <meta itemProp="priceCurrency" content={currency} /> }
    { priceRaw && <meta itemProp="priceCurrency" content={currency} /> }

    <meta itemProp="price" content={priceRaw} />
    <dt>List Price:</dt>
    <dd><b className="price">{price}</b></dd>
    <dt>Trade Price:</dt>
    <dd className="advertise-trade">Trade pricing is reserved for members of our Trade Program. <a href="https://deringhall.com/trade">Apply Now</a></dd>
  </dl>
);

CommerceModulePricing.propTypes = {
  price: PropTypes.string.isRequired,
  currency: (props, propName) => {
    const currency = props[propName];

    if (typeof props.priceRaw === 'number' &&
        (currency === undefined || typeof currency !== 'string')
    ) {
      throw new Error("'currency' is required.");
    }
  },
  priceRaw: PropTypes.number
};

export default CommerceModulePricing;
