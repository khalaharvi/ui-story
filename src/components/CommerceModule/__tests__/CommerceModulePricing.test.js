import React from 'react';
import { shallow } from 'enzyme';

import CommerceModulePricing from '../CommerceModulePricing';

describe('<CommerceModulePricing />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <CommerceModulePricing
        price="Price on Request"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with priceRaw', () => {
    const wrapper = shallow(
      <CommerceModulePricing
        price="$1,000"
        priceRaw={1000}
        currency="USD"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
