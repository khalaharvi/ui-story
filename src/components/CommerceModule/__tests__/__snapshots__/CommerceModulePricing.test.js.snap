// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommerceModulePricing /> renders correctly 1`] = `
<dl
  className="dh-commerce-module-pricing"
>
  <meta
    itemProp="price"
  />
  <dt>
    List Price:
  </dt>
  <dd>
    <b
      className="price"
    >
      Price on Request
    </b>
  </dd>
  <dt>
    Trade Price:
  </dt>
  <dd
    className="advertise-trade"
  >
    Trade pricing is reserved for members of our Trade Program. 
    <a
      href="https://deringhall.com/trade"
    >
      Apply Now
    </a>
  </dd>
</dl>
`;

exports[`<CommerceModulePricing /> renders correctly with priceRaw 1`] = `
<dl
  className="dh-commerce-module-pricing"
>
  <meta
    content="USD"
    itemProp="priceCurrency"
  />
  <meta
    content="USD"
    itemProp="priceCurrency"
  />
  <meta
    content={1000}
    itemProp="price"
  />
  <dt>
    List Price:
  </dt>
  <dd>
    <b
      className="price"
    >
      $1,000
    </b>
  </dd>
  <dt>
    Trade Price:
  </dt>
  <dd
    className="advertise-trade"
  >
    Trade pricing is reserved for members of our Trade Program. 
    <a
      href="https://deringhall.com/trade"
    >
      Apply Now
    </a>
  </dd>
</dl>
`;
